go/array_string/maximum_profit/maximum_profit.go

```diff
@@ -0,0 +1,19 @@
+// Package maxprofit contains the implementation of the
+// maximum profit algorithm with a time complexity O(n) and
+// space complexity of O(1)
+package maxprofit
+
+func maxProfit(prices []int) int {
+	buyPrice := 1<<63 - 1
```

Comment:
You can also use the `math.Max*int*` consts but this requires you to convert the type to int